src/devices_models/devices/common/cost_functions.jl

```diff
@@ -443,51 +490,96 @@ function add_to_cost!(
     resolution = model_resolution(psi_container)
     dt = Dates.value(Dates.Second(resolution)) / SECONDS_IN_HOUR
-    variable_cost = PSY.get_variable(cost_data)
-    variable_cost!(psi_container, spec, component_name, variable_cost)
+    time_steps = model_time_steps(psi_container)
+    initial_time = model_initial_time(psi_container)
+    variable_cost_forecast = get_time_series(psi_container, component, "variable_cost")
+    variable_cost_forecast = map(PSY.VariableCost, variable_cost_forecast)
+    for t in time_steps
+        variable_cost!(psi_container, spec, component_name, variable_cost_forecast[t], t)
+    end
 
     if !isnothing(spec.start_up_cost)
         # Start-up costs
         @debug "start up cost" component_name
         for (st, var_type) in
             enumerate((HotStartVariable, WarmStartVariable, ColdStartVariable))
             var_name = make_variable_name(var_type, spec.component_type)
-            linear_gen_cost!(
-                psi_container,
-                var_name,
-                component_name,
-                start_cost_data[st] * spec.multiplier,
-            )
+            for t in time_steps
+                linear_gen_cost!(
+                    psi_container,
+                    var_name,
+                    component_name,
+                    spec.start_up_cost[st] * spec.multiplier,
+                    t,
+                )
+            end
         end
     end
 
     if spec.has_status_variable
         @debug "no_load cost" component_name
-        linear_gen_cost!(
-            psi_container,
-            make_variable_name(OnVariable, spec.component_type),
-            component_name,
-            PSY.get_no_load(cost_data) * spec.multiplier,
-        )
+        for t in time_steps
+            linear_gen_cost!(
+                psi_container,
+                make_variable_name(OnVariable, spec.component_type),
+                component_name,
+                PSY.get_no_load(cost_data) * spec.multiplier,
+                t,
+            )
+        end
     end
 
     if !isnothing(spec.shut_down_cost)
         @debug "Shut down cost" component_name
-        linear_gen_cost!(
-            psi_container,
-            make_variable_name(StopVariable, spec.component_type),
-            component_name,
-            spec.shut_down_cost(cost_data) * spec.multiplier,
-        )
+        for t in time_steps
+            linear_gen_cost!(
+                psi_container,
+                make_variable_name(StopVariable, spec.component_type),
+                component_name,
+                spec.shut_down_cost(cost_data) * spec.multiplier,
+                t,
+            )
+        end
     end
 
     # Here goes the services bids
+    ancillary_services = PSY.get_ancillary_services(cost_data)
+    for service in ancillary_services
+        forecast_data = get_time_series(psi_container, component, PSY.get_name(service))
+        forecast_data = map(PSY.VariableCost, forecast_data)
```

Comment:
Doesn't your new PSY code obviate the need to do this?

Reply:
Jose's said he plans on adding caching to the `get_time_series` function and then we can update the usage here, originally I had used the methods from PSY